internal/productcatalog/feature_connector.go

```diff
@@ -0,0 +1,151 @@
+package productcatalog
+
+import (
+	"context"
+	"slices"
+
+	"github.com/openmeterio/openmeter/internal/meter"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+type CreateFeatureInputs struct {
+	Name                string             `json:"name"`
+	Key                 string             `json:"key"`
+	Namespace           string             `json:"namespace"`
+	MeterSlug           string             `json:"meterSlug"`
+	MeterGroupByFilters *map[string]string `json:"meterGroupByFilters"`
+}
+
+type FeatureConnector interface {
+	// Feature Management
+	CreateFeature(ctx context.Context, feature CreateFeatureInputs) (Feature, error)
+	// Should just use deletedAt, there's no real "archiving"
+	ArchiveFeature(ctx context.Context, featureID models.NamespacedID) error
+	ListFeatures(ctx context.Context, params ListFeaturesParams) ([]Feature, error)
+	GetFeature(ctx context.Context, featureID models.NamespacedID) (Feature, error)
+}
+
+type FeatureOrderBy string
+
+const (
+	FeatureOrderByCreatedAt FeatureOrderBy = "created_at"
+	FeatureOrderByUpdatedAt FeatureOrderBy = "updated_at"
+)
+
+type ListFeaturesParams struct {
+	Namespace       string
+	IncludeArchived bool
+	Offset          int
+	Limit           int
+	OrderBy         FeatureOrderBy
+}
+
+type DBCreateFeatureInputs struct {
+	Name                string
+	Key                 string
+	Namespace           string
+	MeterSlug           string
+	MeterGroupByFilters *map[string]string
+}
+
+type FeatureDBConnector interface {
+	CreateFeature(ctx context.Context, feature DBCreateFeatureInputs) (Feature, error)
+	ArchiveFeature(ctx context.Context, featureID models.NamespacedID) error
+	ListFeatures(ctx context.Context, params ListFeaturesParams) ([]Feature, error)
+	FindByKey(ctx context.Context, namespace string, key string, includeArchived bool) (*Feature, error)
+	GetByID(ctx context.Context, featureID models.NamespacedID) (Feature, error)
+
+	entutils.TxCreator
+	entutils.TxUser[FeatureDBConnector]
+}
+
+type featureConnector struct {
+	db        FeatureDBConnector
+	meterRepo meter.Repository
+}
+
+func NewFeatureConnector(
+	db FeatureDBConnector,
+	meterRepo meter.Repository,
+) FeatureConnector {
+	return &featureConnector{
+		db:        db,
+		meterRepo: meterRepo,
+	}
+}
+
+func (c *featureConnector) CreateFeature(ctx context.Context, feature CreateFeatureInputs) (Feature, error) {
+	meter, err := c.meterRepo.GetMeterByIDOrSlug(ctx, feature.Namespace, feature.MeterSlug)
+	if err != nil {
+		return Feature{}, &models.MeterNotFoundError{MeterSlug: feature.MeterSlug}
+	}
+
+	validAggregations := []models.MeterAggregation{
```

Comment:
can be a const top of file

Reply:
IMO should be config to the connector, feature doesn't implement any querying, it doesn't really know what aggregation types it can support....

Reply:
Putting aside the theoretical part, usually these are defined as globals as @hekike suggested.